src/parse.rs

```diff
@@ -102,31 +95,59 @@ pub enum ParseResult {
     Unsupported(SentenceType),
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum NmeaError<'a> {
+    /// The provided input was not a proper UTF-8 string
+    Utf8DecodingError,
+    /// The checksum of the sentence was corrupt or wrong
+    ChecksumMismatch,
+    /// For some reason a sentence was passed to the wrong sentence specific parser, this error
+    /// should never happen. First slice is the expected header, second is the found one
+    WrongSentenceHeader(&'a [u8], &'a [u8]),
+    /// The sentence could not be parsed because its format was invalid
+    ParsingError(nom::Err<(&'a [u8], nom::error::ErrorKind)>),
+    /// The sentence was too long to be parsed, our current limit is 102 characters
```

Comment:
I suppose it would be better introduce constant for 102, and reference constant here.
Because of now 102 repeating twice in our code base.

Reply:
Not quite sure what you mean by this? If i git grep through our code base i see
```
parse.rs:    if sentence.len() > 102 {
parse.rs:    /// The sentence was too long to be parsed, our current limit is 102 characters
sentences/gga.rs:/// (empty field) DGPS station ID number (0000-1023)
```
Shall i just put the 102 from the first result into a const and use it there? Or do you mean something else?

Reply:
> Shall i just put the 102 from the first result into a const and use it there?

Yes,
```
const MAX_ : ... = 102;
```
and use at in `if sentence.len() > MAX_...` and in doc comment, with "`": 
 ```
/// The sentence was too long to be parsed `MAX...`